examples/states_ui/welcome.rs

```diff
@@ -0,0 +1,52 @@
+use amethyst::{
+    ecs::prelude::Entity,
+    input::{is_close_requested, is_key_down, is_mouse_button_down},
+    prelude::*,
+    ui::UiCreator,
+    winit::{MouseButton, VirtualKeyCode},
+};
+
+use log::info;
+
+use crate::util::delete_hierarchy;
+
+#[derive(Default, Debug)]
+pub struct WelcomeScreen {
+    ui_handle: Option<Entity>,
+}
+
+impl SimpleState for WelcomeScreen {
+    fn on_start(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        let world = data.world;
+
+        self.ui_handle =
+            Some(world.exec(|mut creator: UiCreator<'_>| creator.create("ui/welcome.ron", ())));
+    }
+
+    fn handle_event(
+        &mut self,
+        _: StateData<'_, GameData<'_, '_>>,
+        event: StateEvent,
+    ) -> SimpleTrans {
+        match &event {
+            StateEvent::Window(event) => {
+                if is_close_requested(&event) || is_key_down(&event, VirtualKeyCode::Escape) {
+                    Trans::Quit
+                } else if is_mouse_button_down(&event, MouseButton::Left) {
+                    info!("Switching to MainMenu!");
```

Comment:
I would recommend doing this for each state transition

Reply:
will do.